pyproject.toml

```diff
@@ -209,15 +208,44 @@ mapserver = [
 ]
 test = [
     "pyaml>=17.10.0",
-    "pytest>=6.1",
+    "pytest>=8.0.1",
     "pytest-mpl>=0.16.1",
-    "sympy!=1.10,!=1.9", # see https://github.com/sympy/sympy/issues/22241
-    "imageio!=2.35.0", # see https://github.com/yt-project/yt/issues/4966
 ]
 
 [project.scripts]
 yt = "yt.utilities.command_line:run_main"
 
+[dependency-groups]
+docs = [
+    "alabaster>=0.7.13",
+    "bottle>=0.12.25",
+    "ipykernel>=6.29.4",
+    "jinja2<3.2.0",
+    "jupyter-client>=8.3.1",
+    "nbsphinx>=0.9.3",
+    "nose~=1.3.7 ; python_full_version < '3.10'",
+    "pytest>=6.1",
+    "pyx>=0.15",
+    "sphinx>=7.2.5",
+    "sphinx-bootstrap-theme>=0.8.1",
+    "sphinx-rtd-theme>=1.3.0",
+]
+geophysics = [
```

Comment:
I'd replace `geophysics` with something more general and mapping-related since cartopy is used purely for 2d plotting and the term "geophysics" has broader meaning. Maybe just `mapping` ?